catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/SolrProviderRealTimeQueryTest.java

```diff
@@ -261,4 +280,42 @@ public void testRealTimeQueryNotId() throws Exception {
     // Verify a "not ID query" does not return the result.
     queryAndVerifyCount(0, filter, provider);
   }
+
+  @Test
+  public void testCommitNrt() throws Exception {
+
+    final String metacardTitle = "testCommitNrt";
+
+    deleteAll(provider);
+
+    MetacardType nrtMetacardType =
+        new MetacardTypeImpl(
+            COMMIT_NRT_TYPE, MetacardImpl.BASIC_METACARD.getAttributeDescriptors());
+
+    MockMetacard metacard = new MockMetacard(Library.getFlagstaffRecord(), nrtMetacardType);
+    metacard.setTitle(metacardTitle);
+
+    CreateResponse createResponse = create(metacard, provider);
+    Optional<String> id =
+        createResponse.getCreatedMetacards().stream().map(Metacard::getId).findFirst();
+
+    if (!id.isPresent()) {
+      Assert.fail("Metacard creation failed for metacard with title: " + metacardTitle);
+    }
+
+    RetryPolicy policy =
+        new RetryPolicy()
+            .retryIf((SourceResponse response) -> response.getResults().size() == 0)
+            .withDelay(100, TimeUnit.MILLISECONDS)
+            .withMaxDuration(30, TimeUnit.SECONDS);
+
+    Query query =
+        new QueryImpl(
+            getFilterBuilder().attribute(Metacard.TITLE).is().equalTo().text(metacardTitle));
+    QueryRequest request = new QueryRequestImpl(query);
+
+    SourceResponse response = Failsafe.with(policy).get(() -> provider.query(request));
+
+    assertThat(response.getResults().size()).isEqualTo(1);
```

Comment:
✏️ Alternatively you could use the `hasSize` matcher

`assertThat(response.getResults(), hasSize(1));`

Reply:
This test is using Google Truth.  I am not sure if I could use that matcher with it.